File: ReSharper.FSharp/src/FSharp.Common/src/ProjectModel/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp.ProjectModel

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.ReSharper.Plugins.FSharp

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<ILanguageFSharpZone>

File: ReSharper.FSharp/src/FSharp.TypeProviders.Protocol/src/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;
using JetBrains.ProjectModel.NuGet;
using JetBrains.Rider.Model;

namespace JetBrains.ReSharper.Plugins.FSharp.TypeProviders.Protocol
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<ILanguageFSharpZone>, IRequire<INuGetZone>, IRequire<IRiderModelZone>
  {
  }
}
